nkms_eth/actors.py

```diff
@@ -0,0 +1,263 @@
+from collections import OrderedDict
+from typing import Tuple, List
+
+# from nkms_eth.agents import MinerAgent, PolicyAgent
+from nkms_eth.base import Actor
+
+
+class PolicyArrangement:
+    def __init__(self, author: 'PolicyAuthor', miner: 'Miner', value: int,
+                 periods: int, arrangement_id: bytes=None):
+
+        if arrangement_id is None:
+            self.id = self.__class__._generate_arrangement_id()  # TODO: Generate policy ID
+
+        # The relationship exists between two addresses
+        self.author = author
+        self.policy_agent = author.policy_agent
+
+        self.miner = miner
+
+        # Arrangement value, rate, and duration
+        rate = value // periods
+        self._rate = rate
+
+        self.value = value
+        self.periods = periods  # TODO: datetime -> duration in blocks
+
+        self.is_published = False
+
+    @staticmethod
+    def _generate_arrangement_id(policy_hrac: bytes) -> bytes:
+        pass  # TODO
+
+    def __repr__(self):
+        class_name = self.__class__.__name__
+        r = "{}(client={}, node={})"
+        r = r.format(class_name, self.author, self.miner)
+        return r
+
+    def publish(self, gas_price: int) -> str:
+
+        payload = {'from': self.author.address,
+                   'value': self.value,
+                   'gas_price': gas_price}
+
+        txhash = self.policy_agent.transact(payload).createPolicy(self.id,
+                                                                    self.miner.address,
+                                                                    self.periods)
+
+        self.policy_agent._blockchain._chain.wait.for_receipt(txhash)
```

Comment:
This blocks until the transaction is published?

Reply:
Yes - http://populus.readthedocs.io/en/latest/chain.wait.html#populus.wait.Wait.for_receipt